### PR TITLE
Isolate the Random loadout logic in its own file

### LINKED ISSUE #836: move applyRandomLoadout out of dimAppCtrl.controller.js
we should clean up dimAppCtrl... the random loadout logic should eventually move to it's own file

also should support the 'revert loadout' or 'before random'

maybe consider moving the logic to where we do the max-light loadout to reuse and get some additional logic for free.

### PR DESCRIPTION
**Pull request by yannickguillemot**
Fixes #836.

This PR simply extract the RandomLoadout controller logic into its own file. This file is stored in the loadout directory and in its own subdirectory to allow for future additions of directives/components.

This is my first contribution and my angular skills are perfectible. Feel free to be good critics.

### CODE REVIEW

**Comment by bhollis** [comment]
Nice! I might tweak this a bit more later, but for now it totally does the trick.

**Comment by yannickguillemot** [comment]
Awesome. Any other comments?

**Comment by SunburnedGoose** [comment]
The only question I have is why wrap the anchor in a div?  Does the div have inline-block?  I haven't pulled this to run it yet.

**Comment by yannickguillemot** [comment]
Yeah. Not being familiar with angular and its binding in views, I tried to keep it as it was before with just the anchor. But to use the newly created random controller, I had to wrap it into a div tag where I define a new controller scope (```ng-controller="dimRandomCtrl as random"```). If there is a cleaner way to do this, I am all for it. 
TL;DR, it was not to solve a design issue, just an angular method binding problem.

**Comment by bhollis** [comment]
Thanks @yannickguillemot!
